py/client-ticking/build.gradle

```diff
@@ -0,0 +1,108 @@
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+    id 'io.deephaven.project.register'
+    id 'io.deephaven.deephaven-in-docker'
+}
+
+evaluationDependsOn(':cpp-client')
+
+configurations {
+    python {}
+}
+
+dependencies {
+    python project(':cpp-client')
```

Comment:
If this does something, we shouldn't need the `evaluationDependsOn` above. With that said, I don't see that cpp-client actually declares any artifacts, and this project doesn't use `configurations.python` for anything to put those artifacts to use, so I think you can safely delete the `configurations` and `dependencies` blocks.

Using artifacts is "better" in that it means gradle can actually see the files moving back and forth, figure out when changes don't have to happen. Other places where we have a `python` configuration are usually when we have a `.whl` file copied out of the container and ready for another task (typically another docker container) to copy in. I think we could use that approach here if the cpp-client:cppClient tasks set up some `copyOut` of compiled cpp output files... but if we're just depending on the docker image produced in cpp-client, `evaluationDependsOn` and `parentContainers` is probably sufficient.

---


Okay, that being said, the `parentContainers` reference below points at py-client, not cpp-client, and py-client points at cpp-client. This means that technically gradle is allow to build this project after cpp-client but before py-client, which will result in an NPE when that parentContainers is evaluated. To fix that, change the `evaluationDependsOn` to point at py-client instead of cpp-client.

Reply:
Fixed, thanks.